modules/php/States/SelectCardTrait.php

```diff
@@ -49,26 +51,19 @@ public function argSelect()
   public function select($ids)
   {
     $args = Log::getLastAction('selection');
-    $selection = Cards::getSelection();
+    $cards = Cards::getSelection();
 
-    // Compute the remeaning cards
-    $rest = [];
-    foreach ($selection['cards'] as $card) {
-      if (!in_array($card['id'], $ids)) {
-        $rest[] = $card['id'];
-      }
-    }
+    // Compute the remaining cards
+    $rest = array_filter($cards, function($card) use ($ids) {
```

Comment:
And now you can use ->filter() to compute the rest since getSelection return a collection instead of an array !